app/src/main/java/com/ddoczi/tasky/agenda/data/remote/uploader/EventUploaderImpl.kt

```diff
@@ -1,13 +1,53 @@
 package com.ddoczi.tasky.agenda.data.remote.uploader
 
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequestBuilder
 import androidx.work.WorkManager
+import com.ddoczi.tasky.agenda.data.mapper.toCreateDto
+import com.ddoczi.tasky.agenda.data.mapper.toUpdateDto
+import com.ddoczi.tasky.agenda.data.remote.dto.CreateEventDto
+import com.ddoczi.tasky.agenda.data.remote.dto.UpdateEventDto
+import com.ddoczi.tasky.agenda.data.remote.worker.EventUploaderWorker
+import com.ddoczi.tasky.agenda.data.remote.worker.EventUploaderWorkerParameters
 import com.ddoczi.tasky.agenda.domain.model.AgendaItem
 import com.ddoczi.tasky.agenda.domain.uploader.EventUploader
+import com.squareup.moshi.Moshi
+import java.util.concurrent.TimeUnit
+import java.time.Duration
 
 class EventUploaderImpl(
     private val workManager: WorkManager
 ) : EventUploader {
     override suspend fun uploadEvent(event: AgendaItem.Event, isEdit: Boolean) {
-        //TODO: Implement uploadEvent
+        val moshi = Moshi.Builder().build()
+        val json: String = if (isEdit) {
+            val jsonAdapter = moshi.adapter(UpdateEventDto::class.java)
+            jsonAdapter.toJson(event.toUpdateDto())
+        } else {
+            val jsonAdapter = moshi.adapter(CreateEventDto::class.java)
+            jsonAdapter.toJson(event.toCreateDto())
+        }
+
+        val uploaderWorker = PeriodicWorkRequestBuilder<EventUploaderWorker>(
+            15, TimeUnit.MINUTES,
+            5, TimeUnit.MINUTES
+        ).setConstraints(
+            Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build()
+        ).setInputData(
+            Data.Builder()
+                .putString(EventUploaderWorkerParameters.EVENT_JSON, json)
+                .putBoolean(EventUploaderWorkerParameters.IS_EDIT, isEdit)
+                .build()
+        ).setBackoffCriteria(BackoffPolicy.EXPONENTIAL, Duration.ofMinutes(0)).build()
```

Comment:
Why a periodic worker to upload a single event? 🤔 
This will upload the same event every 15 minutes